dotnet/src/Connectors/Connectors.Memory.Postgres/PostgresVectorStoreRecordCollection.cs

```diff
@@ -0,0 +1,317 @@
+        {
+            yield return VectorStoreErrorHandler.RunModelConversion(
+                DatabaseName,
+                this.CollectionName,
+                operationName,
+                () => this._mapper.MapFromStorageToDataModel(row, new() { IncludeVectors = includeVectors }));
+        }
+    }
+
+    /// <inheritdoc/>
+    public async Task DeleteAsync(TKey key, DeleteRecordOptions? options = null, CancellationToken cancellationToken = default)
+    {
+        await this._client.DeleteAsync(this.CollectionName, this._propertyReader.KeyPropertyStoragePropertyName, key, cancellationToken).ConfigureAwait(false);
+    }
+
+    /// <inheritdoc/>
+    public Task DeleteBatchAsync(IEnumerable<TKey> keys, DeleteRecordOptions? options = null, CancellationToken cancellationToken = default)
+    {
+        return this._client.DeleteBatchAsync(this.CollectionName, this._propertyReader.KeyPropertyStoragePropertyName, keys, cancellationToken);
+    }
+
+    /// <inheritdoc />
+    public Task<VectorSearchResults<TRecord>> VectorizedSearchAsync<TVector>(TVector vector, VectorSearchOptions? options = null, CancellationToken cancellationToken = default)
+    {
+        Verify.NotNull(vector);
+
+        var vectorType = vector.GetType();
+
+        if (!PostgresConstants.SupportedVectorTypes.Contains(vectorType))
+        {
+            throw new NotSupportedException(
+                $"The provided vector type {vectorType.FullName} is not supported by the SQLite connector. " +
+                $"Supported types are: {string.Join(", ", PostgresConstants.SupportedVectorTypes.Select(l => l.FullName))}");
+        }
+
+        var searchOptions = options ?? s_defaultVectorSearchOptions;
+        var vectorProperty = this.GetVectorPropertyForSearch(searchOptions.VectorPropertyName);
+
+        if (vectorProperty is null)
+        {
+            throw new InvalidOperationException("The collection does not have any vector properties, so vector search is not possible.");
+        }
+
+        var pgVector = PostgresVectorStoreRecordPropertyMapping.MapVectorForStorageModel(vector);
+
+        Verify.NotNull(pgVector);
+
+        // Simulating skip/offset logic locally, since OFFSET can work only with LIMIT in combination
+        // and LIMIT is not supported in vector search extension, instead of LIMIT - "k" parameter is used.
+        var limit = searchOptions.Top + searchOptions.Skip;
+
+        var results = this._client.GetNearestMatchesAsync(
+            this.CollectionName,
+            this._propertyReader.RecordDefinition.Properties,
+            vectorProperty,
+            pgVector,
+            searchOptions.Top,
+            searchOptions.Filter,
+            searchOptions.Skip,
+            searchOptions.IncludeVectors,
+            cancellationToken
+            ).Select(result =>
+            {
+                var record = this._mapper.MapFromStorageToDataModel(
```

Comment:
Remember to wrap mapping errors here as well using VectorStoreErrorHandler.RunModelConversion